backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2NativeCompilerArguments.kt

```diff
@@ -60,8 +62,10 @@ class K2NativeCompilerArguments : CommonCompilerArguments() {
     @Argument(value = "-nopack", description = "Don't pack the library into a klib file")
     var nopack: Boolean = false
 
-    @Argument(value="-linker-options", deprecatedName = "-linkerOpts", valueDescription = "<arg>", description = "Pass arguments to linker", delimiter = " ")
-    var linkerArguments: Array<String>? = null
+    @Argument(value="-linker-options", deprecatedName = "-linkerOpts", valueDescription = "<arg>", description = "Pass arguments to linker", delimiter = ARGUMENT_NO_DELIMITER)
+    var linkerArgumentsInternal: Array<String> = arrayOf()
+
+    val linkerArguments: Array<String> by parseCommandLineString { linkerArgumentsInternal }
```

Comment:
Handling quotes and escaping here makes sense, however quoting arguments in Gradle plugin to unquote them in the compiler doesn't seem to be proper solution to me.
Instead I suggest adding new command-line argument, `-linkerOpt`, which would allow passing next argument to the linker as-is.

This new option should be handled preserving mutual order with the old one, i.e.
```
-linkerOpt arg1 -linkerOpts "arg2 arg3" -linkerOpt "arg 4"
```
->
```
[arg1, arg2, arg3, arg 4]
```

Reply:
What is the best place to write tests for that code? I was looking for the right module for it. Please recommend one to add tests there.

Reply:
btw, the exact case would work. The `@Argument` annotation generate the following array for us:
```
[arg1, "arg2 arg3", "arg 4"]
```
the quotes around the second and the third values are removed by the commanline parser, when you start the process
the artificial `delimiter` value will not let it to split arguments
Next after processing the quotes we will have:
```
[arg1, arg2, arg2, arg, 4]
```
One may need double quotes to pass a `arg 4` as a single argument, e.g. `-linerOps "\"arg 4\""`

Reply:
> The `@Argument` annotation generate the following array for us:

How can it generate this array if `-linkerOpt` is not supported yet?

> One may need double quotes to pass a `arg 4` as a single argument, e.g. `-linerOps "\"arg 4\""`

What's the motivation behind applying unreliable solution to pass data between Gradle plugin and compiler?

Reply:
Quotes in quotes are ok. 

Unreliable? - I suggest to consider a way of passing arguments between processes (Gradle, cinterop, kotlinc-native, klib metadata) in a way, where we do not need to encode/decode arguments many times. The approach is error-prone by definition. That is what looks _reliable_ to me. 

I suggested adding the new `-linkerOpt` argument to _avoid_ us from parsing longer lines of multiple arguments on and on again. It has nothing related to the current change around the `@Argument` annotation.

Reply:
> I suggest to consider a way of passing arguments between processes (Gradle, cinterop, kotlinc-native, klib metadata)

This is *exactly* what I've suggested above, but you've mentioned some problems that don't happen with such a solution:

>  [arg1, arg2, arg2, arg, 4]

> One may need double quotes to pass a `arg 4` as a single argument, e.g. `-linerOps "\"arg 4\""`

Reply:
👍 sorry, I did not spot you used two different options in the very first comment: `-linkerOpt` and `-linkerOpts`. 
I will try addressing that